### PR TITLE
feat(product): add url suffix and URI truncation to Magento driver

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

- magento driver returns products without suffixed URLs
- magento driver queries products with URIs potentially containing suffixes (which doesn't work)

## What is the new behavior?
- magento driver returns products with suffixed URLs
- magento driver truncates request URIs before making the query

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

add `truncateUri` to product driver configs. The recommended setting is `true`.

## Other information

### CODE REVIEW

File: libs/product/driver/magento/src/product-driver.module.ts [hunk 1]
```diff
@@ -21,7 +22,7 @@ import { DAFF_MAGENTO_GET_A_PRODUCT_QUERY_NAME } from './queries/get-product';
   ],
 })
 export class DaffProductMagentoDriverModule {
-  static forRoot(config: DaffProductMagentoDriverConfig): ModuleWithProviders<DaffProductMagentoDriverModule> {
+  static forRoot(config: DaffProductMagentoDriverConfig = MAGENTO_PRODUCT_CONFIG_DEFAULT): ModuleWithProviders<DaffProductMagentoDriverModule> {
```
Comment (lderrickable) [inline libs/product/driver/magento/src/product-driver.module.ts:25]: Shouldn't the default be provided in the token definition?

Comment (griest024) [reply libs/product/driver/magento/src/product-driver.module.ts:25]: I added it to the token def but I did not remove it as a default arg here.

File: libs/product/driver/magento/src/product.service.ts [hunk 2]
```diff
@@ -24,6 +24,8 @@ import {
   transformManyMagentoProducts,
 } from './transforms/product-transformers';
 
+const TRUNCATE_URI_LEADING_PATH_SEGMENTS = /.*\/(?<uri>.*)$/;
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/product.service.ts]: added

**Comment by lderrickable** [comment]
lgtm

**Review by lderrickable — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**